spring-jms/src/main/java/org/springframework/jms/listener/AbstractPollingMessageListenerContainer.java

```diff
@@ -349,13 +346,11 @@ protected boolean doReceiveAndExecute(Object invoker, @Nullable Session session,
 					if (ex instanceof JMSException jmsException) {
 						throw jmsException;
 					}
-				}
-				finally {
+				} finally {
 					if (exposeResource) {
 						TransactionSynchronizationManager.unbindResource(obtainConnectionFactory());
 					}
 					observation.stop();
-					scope.close();
```

Comment:
This is not a neutral change. The scope is now closed before the observation is stopped and the transaction finished. This will change the behavior and will have unintended consequences.
Please revert.